### PR TITLE
feat(quantic): add focus control for facetValue

### PR DESCRIPTION
**Pull request by MSLI-Coveo**
[SVCC-2292](https://coveord.atlassian.net/browse/SVCC-2292) 
Add focus control for facetValue. Focus now will be set to first item in the element after `showMore` or `showLess` is clicked.
FacetValue setFocus

### CODE REVIEW

**Review by erocheleau — CHANGES_REQUESTED** [review]
Simplify the code a little bit if it's possible?

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js [hunk 1]
```diff
@@ -364,10 +378,12 @@ export default class QuanticCategoryFacet extends LightningElement {
 
   showMore() {
     this.facet.showMoreValues();
+    this.facetContentChanged = true;
```
[the remark this answers is not in the document]
Comment (MSLI-Coveo) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js:376]: renderedCallback still needed after testing.

File: packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js [hunk 2]
```diff
@@ -390,10 +401,12 @@ export default class QuanticFacet extends LightningElement {
 
   showMore() {
     this.facet.showMoreValues();
+    this.facetContentChanged = true;
```
[the remark this answers is not in the document]
Comment (MSLI-Coveo) [reply packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js:399]: setFocus needed to be fired inside renderedCallback.

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js [hunk 3]
```diff
@@ -39,11 +39,24 @@ export default class QuanticCategoryFacetValue extends LightningElement {
...
    */
   @api nonActiveParent = false;
+  /**
+   * A function used to set focus to the value.
+   * @api
+   * @type {VoidFunction}
+   */
+  @api setFocus() {
+    // @ts-ignore
+    this.htlmElement.focus();
```
[the remark this answers is not in the document]
Comment (MSLI-Coveo) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js]: new setFocus added

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js [hunk 4]
```diff
@@ -39,11 +39,24 @@ export default class QuanticCategoryFacetValue extends LightningElement {
...
+  }
 
   labels = {
     inLabel,
   };
 
+  get htlmElement() {
+    return this.template.querySelector('div [role="button"]');
+  }
+
```
[the remark this answers is not in the document]
Comment (MSLI-Coveo) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js]: method removed

**Comment by lvu285** [comment]
Can I see the demo?

**Review by erocheleau — APPROVED** [review]

**Pull request merged**